File: server/FunctionalTests/Tests/Rojo04/GetProjectPartitions.py
"""
TheNexusAvenger

Tests that the get project partitions request acts correctly.
"""

import NexusGitFunctionalTest



"""
Tests the project partitions as a standalone request.
"""
class Rojo04TestProjectPartitionsAsStandaloneRequestMultipleFile(NexusGitFunctionalTest.NexusGitFunctionalTest):
    """
    Setup for the test.
    """
    def setupTest(self):
        self.workspace.writeFile("rojo.json","{\"name\": \"Nexus Git Repository\",\"servePort\": 20001,\"partitions\": {\"src\": {\"path\": \"src\",\"target\": \"ReplicatedStorage.NexusGit\"},\"test\": {\"path\": \"test\",\"target\":\"ReplicatedStorage.NexusGitTest\"}}}")

    """
    Runs the test.
    """
    def runTest(self):
        # Wait for it to initialize.
        self.setPortNumber(20001)
        self.waitForInitialization()

        # Send a standalone request and get the response.
        response = self.sendGETRequest("/getprojectpartitions")
        self.assertEqual(response,"{\"src\":\"ReplicatedStorage.NexusGit\",\"test\":\"ReplicatedStorage.NexusGitTest\"}")

File: server/NexusGitTests/Functional/Rojo04/GetProjectPartitions.cs
/*
 * TheNexusAvenger
 * 
 * Tests the GetProjectPartitions request with Rojo 0.4.
 */

using NUnit.Framework;

namespace NexusGitTests.Functional.Rojo04
{
    /*
     * Tests getting the partitions as a standalone request.
     */
    public class TestGetProjectPartitions : BaseFunctionalTest
    {
        /*
         * Sets up the test.
         */
        public override void Setup()
        {
            this.Workspace.WriteFile("rojo.json","{\"name\": \"Nexus Git Repository\",\"servePort\": 20001,\"partitions\": {\"src\": {\"path\": \"src\",\"target\": \"ReplicatedStorage.NexusGit\"},\"test\": {\"path\": \"test\",\"target\":\"ReplicatedStorage.NexusGitTest\"}}}");
        }
        
        /*
         * Runs the test.
         */
        public override void Run()
        {
            // Assert the initial status is correct.
            var response = this.SendGETRequest("/getprojectpartitions");
            Assert.AreEqual(response,"{\"ReplicatedStorage.NexusGit\":true,\"ReplicatedStorage.NexusGitTest\":true}");
        }
    }
}